LZ4/src/H5Zlz4.c

```diff
@@ -143,19 +143,24 @@ static size_t H5Z_filter_lz4(unsigned int flags, size_t cd_nelmts,
             else /* do the decompression */
             {
 #if LZ4_VERSION_NUMBER > 10900
-                int compressedBytes = LZ4_decompress_safe(rpos, roBuf, blockSize, origSize);
+                int compressedBytes = LZ4_decompress_safe(rpos, roBuf, compressedBlockSize, origSize);
+                if(compressedBytes != blockSize)
+                {
+                    printf("decompressed size not the same: %d, != %d\n", compressedBytes, blockSize);
+                    goto error;
+                }
 #else
 #if LZ4_VERSION_NUMBER > 10800
```

Comment:
`LZ4_decompress_safe` existed before `LZ4_VERSION_NUMBER` existed https://github.com/lz4/lz4/commit/64547df2dec94632ca8d5dd42b07c22a169b70b7

So I think the version checks can be safely removed.

Reply:
Yes, it would simplify the code. I am not familiar with the LZ4 evolution, is there any chance that older versions of this function were slower than the current implementation?